mlir/lib/Dialect/SparseTensor/Transforms/SparseTensorCodegen.cpp

```diff
@@ -766,17 +768,22 @@ class SparseTensorAllocConverter
       return success();
     }
 
+    // Construct the dim/lvl sizes and the (unused) dim2lvl/lvl2dim buffers.
```

Comment:
As discussed offline, we probably can even change the alloc/empty cases with assert isIdentity, and use dimsizes directly. I was playing it safe during the migration, since the code is now always going through the same steps. But indeed, cleanup TBD when we are comfortable.